src/main/kotlin/com/sourcegraph/cody/edit/EditUtil.kt

```diff
@@ -91,4 +100,28 @@ object EditUtil {
       color.darker()
     }
   }
+
+  /**
+   * Return true if the currently selected text editor for `project` is visiting a file that is
+   * currently ignored by CodyIgnore.
+   */
+  fun currentlySelectedFileIsIgnored(project: Project?): Boolean {
+    if (project == null || project.isDisposed) {
```

Comment:
Because this may block for 30 seconds, let's add an assertion that we are not on EDT